src/main/java/com/rabbitmq/client/ChannelOptions.java

```diff
@@ -0,0 +1,123 @@
+ *       the confirmation status via {@link java.util.concurrent.CompletableFuture}</li>
+ *   <li><b>maxOutstandingConfirms</b> - Maximum number of unconfirmed messages allowed.
+ *       When this limit is reached, {@link Channel#basicPublishAsync} will block until
+ *       confirmations are received. Set to 0 for unlimited (default)</li>
+ * </ul>
+ * <p>
+ * <b>Example usage:</b>
+ * <pre>{@code
+ * // Enable publisher confirms with automatic tracking and a limit of 100 outstanding
+ * ChannelOptions options = new ChannelOptions(true, true, 100);
+ * Channel channel = connection.createChannel(options);
+ *
+ * // Publish with automatic confirmation tracking
+ * CompletableFuture<Void> future = channel.basicPublishAsync(
+ *     "", queueName, null, "Hello".getBytes()
+ * );
+ * future.join(); // Wait for confirmation
+ * }</pre>
+ *
+ * @see Channel#basicPublishAsync(String, String, com.rabbitmq.client.AMQP.BasicProperties, byte[])
+ * @see Connection#createChannel(ChannelOptions)
+ * @see PublishException
+ */
+public class ChannelOptions {
+    /**
+     * Header name for tracking publish sequence numbers in message properties.
+     * <p>
+     * When publisher confirmation tracking is enabled, this header is automatically
+     * added to each published message with the message's sequence number as the value.
+     * This allows the library to correlate Basic.Return responses with the correct message.
+     */
+    public static final String PUBLISH_SEQUENCE_NUMBER_HEADER = "x-java-pub-seq-no";
+
+    private final boolean publisherConfirmationsEnabled;
+    private final boolean publisherConfirmationTrackingEnabled;
+    private final int maxOutstandingConfirms;
+
+    /**
+     * Create channel options with default values (all features disabled).
+     */
+    public ChannelOptions() {
+        this(false, false, 0);
+    }
+
+    /**
+     * Create channel options for publisher confirmation tracking.
+     * <p>
+     * <b>Important:</b> If {@code publisherConfirmationTrackingEnabled} is true,
+     * then {@code publisherConfirmationsEnabled} must also be true.
+     *
+     * @param publisherConfirmationsEnabled enable publisher confirmations on the channel
+     *        (calls {@link Channel#confirmSelect()} automatically)
+     * @param publisherConfirmationTrackingEnabled enable automatic tracking and awaiting of confirmations
+     *        via {@link Channel#basicPublishAsync} methods
+     * @param maxOutstandingConfirms maximum number of unconfirmed messages allowed before
+     *        {@link Channel#basicPublishAsync} blocks. Set to 0 for unlimited (default).
+     *        This prevents unbounded memory growth when publishing faster than the broker can confirm.
+     * @throws IllegalArgumentException if tracking is enabled but confirmations are not enabled
+     */
+    public ChannelOptions(boolean publisherConfirmationsEnabled,
```

Comment:
I'd suggest a builder-based approach instead of a constructor-based one. The builder has reasonable defaults, it provides a fluent API with short method names and ultimately calls a private constructor.

If we add a new setting in the future, we'll have to create a 4-parameter constructor and keep the 3-parameter one for backward compatibility. We avoid this with the builder approach.